Adding total number of excess events

grdw opened the pull request:
Related: 

- https://github.com/quintel/etsource/pull/1144
- https://github.com/quintel/etmodel/pull/2176

antw commented on lib/merit/excess.rb:
```diff
@@ -2,6 +2,14 @@ module Merit
   class Excess
     include NetLoadHelper
 
+    # Public: total_number_of_events
```
Comment formatting is a bit strange: there's no need to repeat the method name (it's an extra thing to change if the method is later renamed). Merit uses [Tomdoc](http://tomdoc.org/) for method comments, so you could simplify this as:

```ruby
# Public: Returns the total number of excess events.
def total_number_of_events
end
```

... or if the return type isn't obvious:

```ruby
# Public: Determines the total number of excess events.
#
# Returns an integer.
def total_number_of_events
end
```

antw commented on lib/merit/excess.rb:
```diff
@@ -26,8 +34,16 @@ def number_of_events(duration)
 
     private
 
+    # Private: over_producing
```
FYI, in Tomdoc a method is either `Public` or `Internal`. They don't refer to the visibility of the method, but whether or not the method is something which can be called by an external app or library (`Public`), or is `Internal` to the library and must not not be used by a third-party. My understanding is that a Public method may not be changed to accept different parameters or return a different data type without incrementing the (major?) version number of the library, but internal methods may:

> To describe the status of a method, you SHOULD use one of several prefixes:
> 
> Public: Indicates that the method is part of the project's public API. This annotation is designed to let developers know which methods are considered stable. You SHOULD use this to document the public API of your project. This information can then be used along with Semantic Versioning to inform decisions on when major, minor, and patch versions should be incremented.
> 
> Internal: Indicates that the method is part of the project's internal API. These are methods that are intended to be called from other classes within the project but not intended for public consumption.
>
> Deprecated: Indicates that the method is deprecated and will be removed in a future version. You SHOULD use this to document methods that were Public but will be removed at the next major version.

grdw commented:
@antw thanks for explaining Tomdoc. Didn't know that there was a specific format. Thanks for explaining. I fixed the comments for Excess as far as I know. 

I removed the comment for the `over_production` lambda simply because I think that code is pretty self-explanatory.
